thread_pool.c

```diff
@@ -4,18 +4,19 @@
 #include <stdbool.h>
 #include <unistd.h>
 
-#include "task_queue.h"
+#include "task.h"
+#include "queue.h"
 
 
 #define THREAD_POOL_SIZE 5
 
-task_queue_t task_queue;
+queue_t queue;
 pthread_t thread_pool[THREAD_POOL_SIZE];
 
 void *
 thread_start_routine(void *arg) {
     while (true) {
-        task_t *task = task_queue_retrieve(&task_queue);
+        task_t *task = (task_t *)queue_retrieve(&queue);
```

Comment:
As stated in the issue #2, it might be a good idea to introduce the `task_execute()` function here.

Reply:
True, introduced the function in 6fa1c4475112a2bf4b363e4b56f525c696b22111.